libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -1254,12 +1254,35 @@ internal class MapRouteLine(
          * @return either the layer ID if found else a default layer ID
          */
         fun getBelowLayer(layerId: String?, style: Style): String {
+            val layers = style.layers
             return when (layerId.isNullOrEmpty()) {
-                false -> style.layers.firstOrNull { it.id == layerId }?.id
-                true -> style.layers.firstOrNull {
-                    it.id.contains(RouteConstants.MAPBOX_LOCATION_ID) || it is SymbolLayer
-                }?.id
-            } ?: LocationComponentConstants.SHADOW_LAYER
+                // if a desired ID was provided, try to find it
```

Comment:
What about extracting this block of code into a `private` `fun`ction and give it a name based on the comment? This way `getBelowLayer` will be easier to read and understand and comment will become superfluous so it won't be necessary.